wolk_gateway_module/connectivity/connectivity_service.py

```diff
@@ -74,22 +74,22 @@ def connected(self) -> bool:
         :returns: connected
         :rtype: bool
         """
-        pass
+        raise NotImplementedError
 
     @abstractmethod
     def connect(self) -> Optional[bool]:
         """Establish connection with WolkGateway."""
-        pass
+        raise NotImplementedError
 
     @abstractmethod
     def reconnect(self) -> Optional[bool]:
         """Reestablish connection with WolkGateway."""
-        pass
+        raise NotImplementedError
 
     @abstractmethod
     def disconnect(self) -> Optional[bool]:
         """Terminate connection with WolkGateway."""
-        pass
+        raise NotImplementedError
```

Comment:
how to implement error? eg NotImplementedError = "some error" in main module?

Reply:
No, that is the name of the error - NotImplemented

Reply:
builtin exception

Reply:
Is it necessary to raise exceptions? This is controlled exit

Reply:
It's an abstract method, so yes

Reply:
so what

Reply:
Attempting to call an abstract method should not just "pass", but raise an exception. Especially if the method is expected to return a value.

Reply:
You answered it, return a value.

Reply:
Regardless of what is in the body of this method, either `pass` or `raise NotImplementedError` - if this method is not implemented for this class, an exception will be raised as `@abstractmethod` decorator required it to be implemented for this class.

This whole change is here to be more explicit, the body of it won't actually be run at any time.